src/consensus/aft/raft.h

```diff
@@ -102,10 +102,14 @@ namespace aft
     bool is_execution_pending = false;
     std::list<std::unique_ptr<AbstractMsgCallback>> pending_executions;
 
-    // When we receive append entries from a new primary, we should rollback
-    // to the index they're telling us about, and replay from here. However
-    // for complex reasons we want to do this once only, and expect to be
-    // correct without doing this thereafter.
+    // When this node receives append entries from a new primary, it may need to
+    // roll back an uncommitted suffix it holds. It must trust the new primary
+    // to dictate this index, and trust it to be honest about its approximation
+    // of committed state. To minimise the window where a malicious node could
+    // manipulate this, and to avoid retransmission of a suffix already hold,
+    // this node only executes this rollback rollback instruction on the first
```

Comment:
I'm not sure if the double rollback is here for comical effect, but I like it. I'm wary of talking about malicious nodes here and elsewhere in CFT, where we do not make guarantees about them. We rightly take this step because it's the conservative thing to do, and anything else is a bug.